### PR TITLE
RHOAI 2.14 Upstream Sync

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Jira Issues:

- https://issues.redhat.com/browse/RHOAIENG-349
- https://issues.redhat.com/browse/RHOAIENG-12277
- https://issues.redhat.com/browse/RHOAIENG-12985
- https://issues.redhat.com/browse/RHOAIENG-11891

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
thought https://issues.redhat.com/browse/RHOAIENG-349 is already on main from https://github.com/red-hat-data-services/rhods-operator/pull/337

**Comment by VaishnaviHire** [comment]
@ykaliuta Any idea on why the webhook tests might be failing?

**Comment by VaishnaviHire** [comment]
> thought https://issues.redhat.com/browse/RHOAIENG-349 is already on main from #337

It was linked in this PR, that I cherry-picked https://github.com/opendatahub-io/opendatahub-operator/pull/1205 . These changes do not seem to be included

**Comment by ykaliuta** [comment]
> @ykaliuta Any idea on why the webhook tests might be failing?

Hmm, interesting, the test should use repository's manifests where it is enabled.

**Comment by ykaliuta** [comment]
> 
> It actually failing on getting DSCI/DSC. Wondering if there is any timing issue since it fetched by Kind

ok, requireInstalled() has some unexpected changes.

File: docs/DESIGN.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ To deploy ODH components seamlessly, ODH operator will watch two CRDs:
               name: knative-serving
           modelmeshserving:
             managementState: Managed
+          modelregistry:
```
Comment (zdtsw) [inline docs/DESIGN.md:65]: can skip this one in this PR

File: docs/upgrade-testing.md [hunk 2]
```diff
@@ -141,10 +141,10 @@ spec:
     kueue:
       managementState: Managed
     trainingoperator:
-      managementState: Removed
+      managementState: Managed
```
Comment (zdtsw) [inline docs/upgrade-testing.md:144]: can skip change in this file

File: docs/upgrade-testing.md [hunk 3]
```diff
@@ -141,10 +141,10 @@ spec:
...
     workbenches:
       managementState: Managed
     trustyai:
-      managementState: Removed
+      managementState: Managed
```
Comment (zdtsw) [inline docs/upgrade-testing.md:148]: same here

**Comment by VaishnaviHire** [comment]
Thanks @zdtsw @ykaliuta , the tests have passed now

**Review by zdtsw — APPROVED** [review]

**Comment by VaishnaviHire** [comment]
Can either of you add the labels @zdtsw @ykaliuta

**Comment by VaishnaviHire** [comment]
We need to also update manifests in midestream

**Pull request merged**